include/SFML/Graphics/Color.hpp

```diff
@@ -59,36 +60,36 @@ class SFML_GRAPHICS_API Color
     /// \param alpha Alpha (opacity) component (in the range [0, 255])
     ///
     ////////////////////////////////////////////////////////////
-    Color(Uint8 red, Uint8 green, Uint8 blue, Uint8 alpha = 255);
+    constexpr Color(Uint8 red, Uint8 green, Uint8 blue, Uint8 alpha = 255);
 
     ////////////////////////////////////////////////////////////
     /// \brief Construct the color from 32-bit unsigned integer
     ///
     /// \param color Number containing the RGBA components (in that order)
     ///
     ////////////////////////////////////////////////////////////
-    explicit Color(Uint32 color);
+    constexpr explicit Color(Uint32 color);
 
     ////////////////////////////////////////////////////////////
     /// \brief Retrieve the color as a 32-bit unsigned integer
     ///
     /// \return Color represented as a 32-bit unsigned integer
     ///
     ////////////////////////////////////////////////////////////
-    Uint32 toInteger() const;
+    [[nodiscard]] constexpr Uint32 toInteger() const;
 
     ////////////////////////////////////////////////////////////
     // Static member data
     ////////////////////////////////////////////////////////////
-    static const Color Black;       //!< Black predefined color
-    static const Color White;       //!< White predefined color
-    static const Color Red;         //!< Red predefined color
-    static const Color Green;       //!< Green predefined color
-    static const Color Blue;        //!< Blue predefined color
-    static const Color Yellow;      //!< Yellow predefined color
-    static const Color Magenta;     //!< Magenta predefined color
-    static const Color Cyan;        //!< Cyan predefined color
-    static const Color Transparent; //!< Transparent (black) predefined color
+    static Color Black;          //!< Black predefined color
+    static Color White;    //!< White predefined color
+    static Color Red;          //!< Red predefined color
+    static Color Green;        //!< Green predefined color
+    static Color Blue;         //!< Blue predefined color
+    static Color Yellow;     //!< Yellow predefined color
+    static Color Magenta;    //!< Magenta predefined color
+    static Color Cyan;       //!< Cyan predefined color
+    static Color Transparent; //!< Transparent (black) predefined color
```

Comment:
Shouldn't this be constexpr or at least const? (also the comments are not aligned)

Reply:
https://stackoverflow.com/questions/29432283/c-static-constexpr-field-with-incomplete-type/41570881

Reply:
To make it clear for others, `constexpr` would require constants to be defined here, but the `Color` type is not complete at this point so it is not possible.
Using `const` is possible (as it was before) and constants would be defined later, when the `Color` type is complete.
A workaround to use `constexpr` is given in the next commit https://github.com/SFML/SFML/pull/1847/commits/77edd6718a6b90478b8d9340a1c5a04aa29e7a03. It uses an auxiliary struct `ColorInit` implicitly convertible to `Color`.